### PR TITLE
escaped witespaces from dataPath for macos

### PR DESCRIPTION
**Pull request by ironoa**

### CODE REVIEW

File: lib/core/files.js [hunk 1]
```diff
@@ -5,13 +5,14 @@ const ospath = require('ospath');
 const path = require('path');
 const process = require('process');
 const yaml = require('js-yaml');
+const strings = require('./strings')
```
Comment (fgimenez) [inline lib/core/files.js:8]:
```suggestion
const strings = require('./strings');
```

File: lib/core/strings.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
 module.exports = {
   removeSpaces: (str) => {
     return str.trim().replace(/\s/g,'-');
+  },
+  escapeMacosPathSpaces: (str) => {
+    return str.trim().replace( /(\s+)/g, '' )
```
Comment (fgimenez) [inline lib/core/strings.js:6]:
```suggestion
    return str.trim().replace( /(\s+)/g, '' );
```

Comment (fgimenez) [reply lib/core/strings.js:6]: It would be nice to add a small unit test for this new function.

File: lib/core/cmd.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ const { spawn } = require('child_process');
 module.exports = {
   exec: async (command, options={}) => {
     return new Promise((resolve, reject) => {
-      const items = command.split(' ');
+      const items = command.split(' '); //TODO problem, it is splitting even posix compliant path strings such as /Application\ Support/
```
Comment (fgimenez) [inline lib/core/cmd.js:8]: Indeed, thanks for pointing it out, as soon as we migrate to the cmd ts package we will get rid of this issue, see https://github.com/w3f/cmd-ts/blob/master/src/cmd.ts

Comment (ironoa) [inline lib/core/cmd.js:8]: the os library returns something like .../Application Support/... on macos platform.

to be compliant with posix we could either translate it to .../Application\ Support/... or to .../ApplicationSupport/...

this command.split force us to this solution: .../ApplicationSupport/...

File: lib/core/strings.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
 module.exports = {
   removeSpaces: (str) => {
     return str.trim().replace(/\s/g,'-');
+  },
+  removePathSpaces: (str) => {
```
Comment (fgimenez) [inline lib/core/strings.js:5]: much better this name :100:

File: test/lib/core/files.js [hunk 5]
```diff
@@ -58,7 +59,7 @@ describe('Files', () => {
 
   describe('dataPath', () => {
     it('should return the data path', () => {
-      const expected = path.join(ospath.data(), 'polkadot-deployer');
+      const expected = strings.removePathSpaces(path.join(ospath.data(), 'polkadot-deployer'));
```
Comment (fgimenez) [inline test/lib/core/files.js:62]: Why do we need this here? as i understand it the new function is not called on each of the subject functions, is that correct?

File: lib/core/strings.js (hunk 4 above)
Comment (ironoa) [reply lib/core/strings.js:5]: I generally love to care about namings....

File: test/lib/core/files.js (hunk 5 above)
Comment (ironoa) [reply test/lib/core/files.js:62]: I could do definitely something better, let me show you in a sec

Comment (ironoa) [reply test/lib/core/files.js:62]: but still we need it, and yes you are right: it is not called on each of them, but each of them are relying on the same this.config.dataPath = strings.removePathSpaces(ospath.data());

**Pull request merged**